modules/ding_nodelist/ding_nodelist.module

```diff
@@ -0,0 +1,2477 @@
+  $conf = $form_state['conf'];
+
+  // Show only allowed content types.
+  $node_types = variable_get('ding_nodelist_node_types', array());
+  $all_node_types = node_type_get_names();
+  $node_types_opts = array();
+  foreach ($node_types as $nt => $active) {
+    if ($active) {
+      $node_types_opts[$nt] = $all_node_types[$nt];
+    }
+  }
+
+  if (empty($node_types_opts)) {
+    drupal_set_message(t('No eligible content types were found, please configure a few first.'), 'error', FALSE);
+  }
+
+  $node_type = isset($conf['content_type']) ? $conf['content_type'] : current(array_keys($node_types));
+
+  $default_conf = array(
+    'content_type' => $node_type,
+    'widget_type' => NL_WIDGET_CAROUSEL,
+    'sort_field' => 'created',
+    'limit' => 5,
+    'more_text' => t('More'),
+    'more_link' => '',
+    'teaser_length' => 100,
+    'autoscroll_delay' => 3000,
+  );
+
+  $form['#prefix'] = '<div id="ding_nodelist-form-wrapper">';
+  $form['#suffix'] = '</div>';
+
+  // Initial settings.
+  $form['ding_nodelist_wrapper'] = array(
+    '#type' => 'fieldset',
+    '#collapsible' => FALSE,
+    '#tree' => FALSE,
+    '#title' => t('Configure nodelist'),
+    '#collapsible' => FALSE,
+  );
+
+  $form['ding_nodelist_wrapper']['content_type'] = array(
+    '#type' => 'select',
+    '#title' => t('Content type'),
+    '#required' => TRUE,
+    '#default_value' => $node_type,
+    '#options' => $node_types_opts,
+    '#description' => t("Select the content type of the list's items."),
+    '#ajax' => array(
+      'wrapper' => 'ding_nodelist-selected-nodes',
+      'callback' => 'ding_nodelist_js',
+      'method' => 'replace',
+      'effect' => 'fade',
+    ),
+  );
+
+  // Set up all the available view modes for this type of nodes.
+  $entity = entity_get_info('node');
+  $node_formats = array();
+  foreach ($entity['view modes'] as $view_mode => $settings) {
+    $node_formats[$view_mode] = $settings['label'];
+  }
+
+  $widget_types = _ding_nodelist_get_widget_types($node_type);
+  $widget_type = isset($conf['widget_type']) ? $conf['widget_type'] : $default_conf['widget_type'];
+  $form['ding_nodelist_wrapper']['widget_type'] = array(
+    '#type' => 'select',
+    '#title' => t('List type'),
+    '#required' => TRUE,
+    '#default_value' => $widget_type,
+    '#options' => $widget_types,
+    '#empty_option' => t('- Select -'),
+    '#description' => t('If single item selected, only the first item will be rendered'),
+    '#ajax' => array(
+      'wrapper' => 'ding_nodelist-item-template',
+      'callback' => 'ding_nodelist_js_template',
+      'method' => 'replace',
+      'effect' => 'fade',
+    ),
+    '#prefix' => '<div class=ding_nodelist-widget-config>',
+  );
+
+  $form['ding_nodelist_wrapper']['widget_dependent'] = array(
+    '#type' => 'container',
+    '#prefix' => '<div id="ding_nodelist-item-template">',
+    '#suffix' => '</div></div>',
+  );
+
+  $autoscrolling_widgets = array(NL_WIDGET_CAROUSEL, NL_WIDGET_HOR_ACCORDION);
```

Comment:
You have separate functions for retrieving widgets with titles and trims. Is there a reason why this pattern is not followed here?